src/Microsoft.DotNet.Build.Tasks/CleanupVSTSAgent.cs

```diff
@@ -78,48 +79,69 @@ private void KillStaleProcesses()
                     }
                 }
             }
         }
 
         private void ReportDiskUsage()
         {
-            string drive = Path.GetPathRoot(AgentDirectory);
-            DriveInfo driveInfo = new DriveInfo(drive);
-            Console.WriteLine("Disk Usage Report");
-            Console.WriteLine($"  Agent directory: {AgentDirectory}");
-            Console.WriteLine($"  Drive letter: {drive}");
-            Console.WriteLine($"  Total disk size: {string.Format("{0:N0}", driveInfo.TotalSize)} bytes");
-            Console.WriteLine($"  Total disk free space: {string.Format("{0:N0}", driveInfo.TotalFreeSpace)} bytes");
-
-            var workingDirectories = Directory.GetDirectories(Path.Combine(AgentDirectory, "_work"));
-            var totalWorkingDirectories = workingDirectories != null ? workingDirectories.Length : 0;
-
-            Console.WriteLine("  Agent info");
-            Console.WriteLine($"    Total size of agent directory: {string.Format("{0:N0}", GetDirectoryAttributes(AgentDirectory).Item1)} bytes");
-            Console.WriteLine($"    Total agent working directories: {totalWorkingDirectories}");
-
-            if (totalWorkingDirectories > 0)
+            string lastDirectoryChecked = AgentDirectory;
+            try
             {
-                int nameLength = 0;
-                foreach(string directoryName in workingDirectories)
+                string drive = Path.GetPathRoot(AgentDirectory);
+                DriveInfo driveInfo = new DriveInfo(drive);
+                Log.LogMessage("Disk Usage Report");
+                Log.LogMessage($"  Agent directory: {AgentDirectory}");
+                Log.LogMessage($"  Drive letter: {drive}");
+                Log.LogMessage($"  Total disk size: {string.Format("{0:N0}", driveInfo.TotalSize)} bytes");
+                Log.LogMessage($"  Total disk free space: {string.Format("{0:N0}", driveInfo.TotalFreeSpace)} bytes");
+
+                var workingDirectories = Directory.GetDirectories(Path.Combine(AgentDirectory, "_work"));
+                var totalWorkingDirectories = workingDirectories != null ? workingDirectories.Length : 0;
+
+                Log.LogMessage("  Agent info");
+                Log.LogMessage($"    Total size of agent directory: {string.Format("{0:N0}", GetDirectoryAttributes(AgentDirectory).Item1)} bytes");
+                Log.LogMessage($"    Total agent working directories: {totalWorkingDirectories}");
+
+                if (totalWorkingDirectories > 0)
                 {
-                    nameLength = directoryName.Length > nameLength ? directoryName.Length : nameLength;
+                    int nameLength = 0;
+                    foreach (string directoryName in workingDirectories)
+                    {
+                        nameLength = directoryName.Length > nameLength ? directoryName.Length : nameLength;
```

Comment:
You could probably use `nameLength` value to avoid hitting the "PathTooLongException".  If we weren't trying to catch `UnauthorizedAccessException`, I'd feel more strongly about preferring that over the try-catch.